hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/rpc/RpcForwarder.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.backend.store.raft.rpc;
+
+import static com.baidu.hugegraph.backend.store.raft.RaftSharedContext.WAIT_RPC_TIMEOUT;
+
+import java.util.concurrent.ExecutionException;
+
+import org.slf4j.Logger;
+
+import com.alipay.sofa.jraft.Status;
+import com.alipay.sofa.jraft.core.NodeImpl;
+import com.alipay.sofa.jraft.entity.PeerId;
+import com.alipay.sofa.jraft.error.RaftError;
+import com.alipay.sofa.jraft.rpc.RaftClientService;
+import com.alipay.sofa.jraft.rpc.RpcResponseClosure;
+import com.alipay.sofa.jraft.util.Endpoint;
+import com.baidu.hugegraph.backend.BackendException;
+import com.baidu.hugegraph.backend.store.raft.RaftClosure;
+import com.baidu.hugegraph.backend.store.raft.RaftNode;
+import com.baidu.hugegraph.backend.store.raft.StoreClosure;
+import com.baidu.hugegraph.backend.store.raft.StoreCommand;
+import com.baidu.hugegraph.backend.store.raft.rpc.RaftRequests.CommonResponse;
+import com.baidu.hugegraph.backend.store.raft.rpc.RaftRequests.StoreCommandRequest;
+import com.baidu.hugegraph.backend.store.raft.rpc.RaftRequests.StoreCommandResponse;
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.Log;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Message;
+import com.google.protobuf.ZeroByteStringHelper;
+
+public class RpcForwarder {
+
+    private static final Logger LOG = Log.logger(RpcForwarder.class);
+
+    private final PeerId nodeId;
+    private final RaftClientService rpcClient;
+
+    public RpcForwarder(RaftNode node) {
+        this.nodeId = node.node().getNodeId().getPeerId();
+        this.rpcClient = ((NodeImpl) node.node()).getRpcService();
+        E.checkNotNull(this.rpcClient, "rpc client");
+    }
+
+    public void forwardToLeader(PeerId leaderId, StoreCommand command,
+                                StoreClosure closure) {
+        E.checkNotNull(leaderId, "leader id");
+        E.checkState(!leaderId.equals(this.nodeId),
+                     "Invalid state: current node is the leader, there is " +
+                             "no need to forward the request");
+        LOG.debug("The node {} forward request to leader {}",
+                  this.nodeId, leaderId);
+
+        StoreCommandRequest.Builder builder = StoreCommandRequest.newBuilder();
+        builder.setType(command.type());
+        builder.setAction(command.action());
+        builder.setData(ZeroByteStringHelper.wrap(command.data()));
+        StoreCommandRequest request = builder.build();
+
+        RpcResponseClosure<StoreCommandResponse> responseClosure;
+        responseClosure = new RpcResponseClosure<StoreCommandResponse>() {
```

Comment:
replace \<StoreCommandResponse\> with \<\>

Reply:
it will compile failed